java/ql/lib/semmle/code/java/security/HardcodedCredentialsComparison.qll

```diff
@@ -0,0 +1,25 @@
+/**
+ * Provides classes and predicates to detect comparing a parameter to a hard-coded credential.
+ */
+
```

Comment:
Oh, and a little NIT: maybe we should have `import java` here for consistency. I guess the imports should be `private` too, but this doesn't seem like it's going to be used anywhere except in the query itself, so 🤷

Reply:
Added the imports. Re: who should cover this, my impression was the involvement of dataflow analysis makes it appropriate for CodeQL, but it's possible I'm underestimating what secret scanning can achieve.

Reply:
Well, that makes sense actually! 👍